test/nginx_opentracing_test.py

```
@@ -91,6 +125,16 @@ def _logEnvironment(self):
             os.path.join(test_log, "nginx-error.log"),
         )
 
+    def _setup_grpc(self):
+        """Set up gRPC connection only when needed"""
+        if self.grpcConn is None:
+            self.grpcConn = grpc.insecure_channel("localhost:8081")
+            try:
+                grpc.channel_ready_future(self.grpcConn).result(timeout=10)
+            except grpc.FutureTimeoutError:
+                raise RuntimeError("Error connecting to gRPC server")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)